### PR TITLE
Created .builds files for building libraries 

### PR DESCRIPTION
**Pull request by dsgouda**
Created .builds files for building libraries instead of .*proj files, modified dirs.proj to build via .builds files
Linked to https://github.com/dotnet/corefx/pull/4743 
@weshaggard

### CODE REVIEW

File: dir.traversal.targets [hunk 1]
```diff
@@ -99,5 +99,8 @@
   <Target Name="Rebuild" DependsOnTargets="Clean;Build" />
 
   <Target Name="RestorePackages" DependsOnTargets="$(TraversalRestorePackagesDependsOn)" />
+  <Target Name="BuildAndTest" DependsOnTargets="Build;Test" />
```
Comment (mellinoe) [inline dir.traversal.targets]: Was this moved from somewhere else? I thought these are already defined somewhere

**Comment by mellinoe** [comment]
Are we going to be separately going through and changing these?
```
<AdditionalProperties>TargetGroup=</AdditionalProperties>
```

Doesn't this mean that right now it will just build the default TargetGroup right now, no matter what you specify to MSBuild?

File: dir.traversal.targets (hunk 1 above)
Comment (weshaggard) [inline dir.traversal.targets]: These were copied from <root>\dirs.targets as we needed them for our new traversal .builds projects.

**Comment by weshaggard** [comment]
@mellinoe `TargetGroup=` is the default and should be a no-op right now. We don't expect folks to be passing in TargetGroup property from a root build and expect them to only show up in the .builds files or perhaps someone can pass them to an individual csproj to test build just that target.

**Comment by weshaggard** [comment]
LGTM @dsgouda please squash the commits.

**Comment by mellinoe** [comment]
Okay I see. TargetGroup is referring to the target framework / runtime. And all of these are CoreCLR projects (for now), so that makes sense assuming it's the default.

**Comment by davidsh** [comment]
cc: @stephentoub @CIPop @ericstj 

Why are we doing this?  Will this break the Visual Studio build/debug/test experience? Where/how are we documenting this new `.builds` file for all of the libraries?

**Comment by mellinoe** [comment]
@davidsh There is a bit more context and documentation in https://github.com/dotnet/corefx/pull/4743.

It shouldn't affect the VS build/debug/test experience at all. Basically these .builds files are going to be helper wrappers over project files so that you can easily build a source project multiple times for all valid configurations.

For example, System.Runtime will have an extra set of configurations for .NET Native. If you open the solution in VS, there will just be two more configuration options in the drop down (netcore50_Debug and netcore50_Release), but everything else should otherwise be the same.

**Comment by weshaggard** [comment]
@mellinoe is correct that they are convenience wrappers for different target runtimes or frameworks and should not impact the ability to work with the main projects in VS. However I want to correct one statement which is the .builds files will build for one of the combinations (defaulting to windows) if you want to build for the other OS's you need to pass /p:OSGroup=<OS> to the .builds file. 

We will be doing a number of other library ports over the next few weeks following this new convention but please have a read over https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md and provide feedback as necessary.

**Pull request merged**